markdownlint/TOP002_noCodeInHeadings/TOP002_noCodeInHeadings.js

```diff
@@ -10,9 +10,17 @@ module.exports = {
     );
 
     headingsWithCode.forEach((heading) => {
+      const codeContent = heading.line.match(/`.+?`/);
+
       onError({
         lineNumber: heading.lineNumber,
         detail: `Headings should not contain inline code.`,
+        fixInfo: {
+          lineNumber: heading.lineNumber,
+          editColumn: codeContent.index + 1,
+          deleteCount: codeContent[0].length,
+          insertText: codeContent[0].replace(/`/g, ""),
```

Comment:
Per the markdownlint docs, all of these are optional but at least deleteCount or insertText should be included.

editColumn essentially sets where the deleteCount and insertText properties delete/insert. Without it it would delete/insert at the start of the line. It's also 1-based, hence the `+ 1`

Reply:
From what I have tried, the `lineNumber` and `editColumn` are unnecessary in this case.

Reply:
lineNumber no, we could omit that in this instance. editColumn however is needed. Without it, if the context is:

```
### Some `inline` code
```

Then the fixer outputs:

```
inline `inline` code
```

Rather than what we want:

```
### Some inline code
```

Reply:
Oh, I see. I was confounding this rule with rule TOP003, where the fix simply replaces the entire thing.

Reply:
Considered if just replacing the whole heading would have simpler code (can omit `codeContent` and `fixInfo.editColumn` as well), but turns out it interferes with fixing MD022 (blanks around headings) and doesn't fix with a preceding blank line if that's needed :facepalm: